### PR TITLE
Filter records where SubscriberKey is NULL

### PR DESCRIPTION
**Pull request by drdee**
This addresses the issue where sometimes the ET API returns records where SubscriberKey is NULL. 

This supersedes https://github.com/singer-io/tap-exacttarget/pull/32 and https://github.com/singer-io/tap-exacttarget/pull/27

### CODE REVIEW

**Comment by drdee** [comment]
Fixes #26

File: tap_exacttarget/endpoints/events.py [hunk 1]
```diff
@@ -87,6 +87,13 @@ def sync_data(self):
                                              'EventDate',
                                              event.get('EventDate'))
 
+                        if event.get('SubscriberKey') is None:
+                            LOGGER.info("SubscriberKey is NULL so ignoring {} record with SendID: {} and EventDate: {}"
+                                    .format(event_name,
+                                            event.get('SendID'),
+                                            event.get('EventDate')))
+                        continue
```
Comment (cosmok) [inline tap_exacttarget/endpoints/events.py]: should be indented to align with LOGGER.info

Comment (drdee) [reply tap_exacttarget/endpoints/events.py]: Fixed

Comment (cosmok) [reply tap_exacttarget/endpoints/events.py]: hmm, i meant the continue should be aligned...

**Comment by cosmok** [comment]
@luandy64 , @cmerrick  and @dmosorast or anyone from Stitch, please review and merge it in to create a new version.

Our Stitch integration is broken due to this issue and would be great if it could be resolved.

**Comment by cmerrick** [comment]
Planning to move forward with this approach based on the insight from #27 that the ExactTarget automation studio uses this logic. @cosmok did you ever hear back from the Marketing Cloud folks on the case you opened about this?

**Comment by cosmok** [comment]
Hi @cmerrick thanks for checking this. Exacttarget have created an internal task, there is no timeframe given by them. And they are yet to decide if this behaviour would just be documented or a fix would be applied to not return the entries with null subscriberkey.

**Comment by cmerrick** [comment]
Moving forward with this as-is, @cosmok if you can update this issue if/when that case gets updated, we would appreciate it!

**Pull request merged**